DemoInfo/DemoParser.cs

```diff
@@ -506,9 +527,142 @@ public void ParseHeader()
+		{
+			ExpressParseToEnd();
+			Header.PlaybackTicks = IngameTick;
+			Header.PlaybackFrames = CurrentTick - (TickAtStart + 1);
+			Header.PlaybackTime = TickInterval * IngameTick;
+			
+			CurrentTick = 0;
+
+			//the ol' switcheroo
+			//It's probably safest and easiest to
+			//swap out for a fresh bitstream.
+			Input.Seek(0, System.IO.SeekOrigin.Begin);
+			BitStream = BitStreamUtil.Create(Input);
+			//Put the new stream back at the end of the header
+			ExpressParseHeader(BitStream);
+		}
+
+		//These express functions mirror functions without the express prefix,
+		//but all they do is read ticks and get ServerInfo.  They do not parse
+		//data from any other packets.
+		void ExpressParseToEnd()
+		{
+			while (true)
+			{
+				try
+				{
+					if (!ExpressParseTick())
+						break;
+				}
+				catch (System.IO.EndOfStreamException)
+				{
+					break;
+				}
+			}
+		}
+
+
+		bool ExpressParseTick()
+		{
+			DemoCommand command = (DemoCommand)BitStream.ReadByte();
+
+			IngameTick = (int)BitStream.ReadInt(32); // tick number
+
+			if (IngameTick == 0)
+				TickAtStart = CurrentTick;
+
+			BitStream.ReadByte(); // player slot
+
+			this.CurrentTick++; // = TickNum;
+
+			switch (command)
+			{
+				case DemoCommand.Synctick:
+					break;
+				case DemoCommand.Stop:
+					return false;
+				case DemoCommand.ConsoleCommand:
+					BitStream.BeginChunk(BitStream.ReadSignedInt(32) * 8);
+					BitStream.EndChunk();
+					break;
+				case DemoCommand.DataTables:
+					BitStream.BeginChunk(BitStream.ReadSignedInt(32) * 8);
+					BitStream.EndChunk();
+					break;
+				case DemoCommand.StringTables:
+					BitStream.BeginChunk(BitStream.ReadSignedInt(32) * 8);
+					BitStream.EndChunk();
+					break;
+				case DemoCommand.UserCommand:
+					BitStream.ReadInt(32);
+					BitStream.BeginChunk(BitStream.ReadSignedInt(32) * 8);
+					BitStream.EndChunk();
+					break;
+				case DemoCommand.Signon:
+				case DemoCommand.Packet:
+					ExpressParseDemoPacket();
+					break;
+				default:
+					throw new Exception("Can't handle Demo-Command " + command);
+			}
+
+			return true;
+		}
+
+		void ExpressParseDemoPacket()
+		{
+			//Read a command-info. Contains no really useful information afaik. 
+			CommandInfo.Parse(BitStream);
+			BitStream.ReadInt(32); // SeqNrIn
+			BitStream.ReadInt(32); // SeqNrOut
+
+			BitStream.BeginChunk(BitStream.ReadSignedInt(32) * 8);
+			ExpressParsePacket(BitStream, this);
+			BitStream.EndChunk();
+		}
+
+		void ExpressParsePacket(IBitStream bitstream, DemoParser demo)
+		{
+			while (!bitstream.ChunkFinished)
+			{
+				int cmd = bitstream.ReadProtobufVarInt(); //What type of packet is this?
+				int length = bitstream.ReadProtobufVarInt(); //And how long is it?
+				bitstream.BeginChunk(length * 8); //read length bytes
+				if (cmd == (int)SVC_Messages.svc_ServerInfo)
+				{
+					new ServerInfo().Parse(bitstream, demo);
+				}
+				bitstream.EndChunk();
+			}
+		}
+
+		void ExpressParseHeader(IBitStream reader)
```

Comment:
I wasn't sure if it was really as simple as just adding up the number of bits in the Parseheader code or if there was more going on underneath the surface.  It sounds like you're saying there isn't so I'll go ahead and chunk it.